packages/core/contracts-ovm/insured-bridge/implementation/OVM_BridgeDepositBox.sol

```diff
@@ -1,47 +1,249 @@
 
-    event Deposit(
-        uint256 transferId,
+    // Track the total number of deposits. Used as a unique identifier for bridged transfers.
+    uint256 public numberOfDeposits;
+
+    struct WhitelistedToken {
+        address l1Token;
+        uint64 lastBridgeTime;
+    }
+
+    // Mapping of whitelisted L2Token to WhitelistedToken. Contains L1 TokenAddress and the last time this token type
+    // was bridged. Used to rate limit bridging actions to prevent DOS on L1.
+    mapping(address => WhitelistedToken) public whitelistedTokens;
+
+    // Minimum time that must elapse between bridging actions for a given token. Used to rate limit bridging back to L1.
+    uint64 public minimumBridgingDelay;
+
+    /****************************************
+     *                EVENTS                *
+     ****************************************/
+
+    event SetWithdrawContract(address newL1WithdrawContract);
+    event SetMinimumBridgingDelay(uint64 newMinimumBridgingDelay);
+    event WhitelistToken(address l1Token, address l2Token, uint64 lastBridgeTime);
+    event DepositsEnabled(bool enabledResultantState);
+    event FundsDeposited(
+        uint256 depositId,
         uint256 timestamp,
         address sender,
         address recipient,
-        address originToken,
+        address l1Token,
         uint256 amount,
-        uint256 networkId,
-        uint256 maxFee
+        uint256 maxFeePct
     );
-    event WithdrawalContractSet(address withdrawalContract);
-    event TokenWhitelisted(address originToken, address destinationToken);
-    event DepositContractEnabledStateToggled(bool enabled);
+    event TokensBridged(address l2Token, uint256 numberOfTokensBridged, uint256 l1Gas, address caller);
+
+    /****************************************
+     *               MODIFIERS              *
+     ****************************************/
 
-    constructor(address _l2CrossDomainMessenger, address _l1Owner) OVM_CrossDomainEnabled(_l2CrossDomainMessenger) {
-        l1Owner = _l1Owner;
+    modifier onlyIfDepositsEnabled() {
+        require(depositsEnabled, "Contract is disabled");
+        _;
     }
 
-    // Admin functions
+    /**
+     * @notice Construct the OVM Bridge Deposit Box
+     * @param _bridgeRouter Address of the bridge withdraw contract on L1. Tokens are sent to this address at the
+     *          conclusion of fast relay. This address is the "owner" of the deposit box.
+     * @param _minimumBridgingDelay Minimum second that must elapse between L2->L1 token transfer to prevent dos.
+     * @param timerAddress Timer used to synchronize contract time in testing. Set to 0x000... in production.
+     */
+    constructor(
+        address _bridgeRouter,
+        uint64 _minimumBridgingDelay,
+        address timerAddress
+    ) OVM_CrossDomainEnabled(Lib_PredeployAddresses.L2_CROSS_DOMAIN_MESSENGER) OVM_Testable(timerAddress) {
+        require(_bridgeRouter != address(0), "Bad bridge router address");
```

Comment:
Is having the following duplicate code block preferable to calling an internal helper method? Considering bytecode, gas costs, and readability.

```
require(_bridgeRouter != address(0), "Bad bridge router address");
bridgeRouter = _bridgeRouter;
```

Reply:
You'd also want to emit an event `SetBridgeRouter`

Reply:
yup. done. did the same for minimumBridgeDelay